File: 2244-minimum-rounds-to-complete-all-tasks/2244-minimum-rounds-to-complete-all-tasks.cpp
class Solution {
public:
    int minimumRounds(vector<int>& tasks) {
        map<int, int> mp;
        
        for(auto num: tasks){
            mp[num]++;
        }
        
        int freq, ans = 0;
        for(auto i: mp){
            freq = i.second;
            
            if(freq == 1) return -1;
            
            else if(freq % 3 == 0) ans += freq/3;
            
            else{
                ans += freq/3 + 1;
            }
        }
        
        return ans;
    }
};

File: 2244-minimum-rounds-to-complete-all-tasks/2244-minimum-rounds-to-complete-all-tasks.py
class Solution:
    def minimumRounds(self, tasks: List[int]) -> int:
        
        freq, ans = {}, 0
        
        for item in tasks:
            freq[item] = freq.get(item, 0) + 1
        
        for item in freq:
            if freq[item] != 1:
                ans += (freq[item] + 2 )//3
            else:
                return -1
            
        return ans